signature/policy_eval_signedby.go

```diff
@@ -70,6 +71,13 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(image types.UnparsedImage, sig [
 			return PolicyRequirementError(fmt.Sprintf("Signature by key %s is not accepted", keyIdentity))
 		},
 		validateSignedDockerReference: func(ref string) error {
+			r := image.Reference().DockerReference()
+			if _, isCanonical := r.(reference.Canonical); isCanonical || reference.IsNameOnly(r) {
+				if !pr.UntaggedReference.matchesDockerReference(image, ref) {
```

Comment:
> Per IRC, perhaps referencesByDigest would be both more explicit and clearer?

```
            {
                "type": "signedBy",
                "keyType": "GPGKeys",
                "keyPath": "/home/amurdaca/mykey.asc",
                "referenceByDigest": {
                    "type": "prohibit|matchAnyTag|matchPrecisely"
                }
            }
```
I'm not sure what will really happen when both `referenceByDigest` and `signedIdentity` is present, let's see:

* `referenceByDigest == prohibit`: users can pull by tag _only_ as they're used to and `signedIdentity` applies
* `referenceByDigest == matchAnyTag`: users can pull by both tag and digest and pull-by-digest works iff digest-in-signature == user's requested digest. **What should we do now with `signedIdentity`**? see https://github.com/containers/image/pull/129#discussion_r83220097
* `referenceByDigest == matchPrecisely`: users cal pull by both tag and digest but .... *what?*